pystac_client/client.py

```diff
@@ -504,3 +503,25 @@ def get_search_link(self) -> Optional[pystac.Link]:
             ),
             None,
         )
+
+    def _get_collections_href(self, id: Optional[str] = None) -> str:
+        self_href = self.get_self_href()
+        if self_href is None:
+            data_link = self.get_single_link("data")
+            if data_link is None:
+                raise ValueError(
+                    "cannot build a collections href without a self href or a data link"
+                )
+            else:
+                collections_href = data_link.href
+        elif self_href.endswith("/"):
+            collections_href = f"{self_href}collections"
+        else:
+            collections_href = f"{self_href}/collections"
+
+        if id is None:
+            return collections_href
+        elif collections_href.endswith("/"):
+            return f"{collections_href}{id}"
+        else:
+            return f"{collections_href}/{id}"
```

Comment:
To avoid changing current behavior, I only fall back to the data link _if_ there isn't a self href set. Configuring which method to prefer is in the purview of "hueristic configuration", which has been discussed previously, e.g. here: https://github.com/stac-utils/pystac-client/issues/320#issuecomment-1255202184.

Reply:
So in that case -- nothing we can really do to build a link. We could error if the data link isn't absolute?

Reply:
Yes, that's probably your only way out

Reply:
Implemented.